### ISSUE TITLE
Code numbering on by default?

### ISSUE THREAD

**Issue by rowanc1** [executablebooks/mystjs#248]
Have got two users confused by how to turn code numbering on, and expecting it!

You can turn it on in [the frontmatter](https://myst-tools.org/sandbox?tab=demo&myst=LQAtAC0ACgBuAHUAbQBiAGUAcgBpAG4AZwA6AAoAIAAgAGMAbwBkAGUAOgAgAHQAcgB1AGUACgAtAC0ALQAKAAoAYABgAGAAewBjAG8AZABlAC0AYgBsAG8AYwBrAH0AIABwAHkAdABoAG8AbgAKADoAbgBhAG0AZQA6ACAAbQB5AC0AcAByAG8AZwByAGEAbQAKADoAYwBhAHAAdABpAG8AbgA6ACAAQwByAGUAYQB0AGkAbgBnACAAYQAgAFQAZQBuAHMAbwByAE0AZQBzAGgAIAB1AHMAaQBuAGcAIABTAGkAbQBQAEUARwAKAGYAcgBvAG0AIABkAGkAcwBjAHIAZQB0AGkAegBlACAAaQBtAHAAbwByAHQAIABUAGUAbgBzAG8AcgBNAGUAcwBoAAoACgBoAHgAIAA9ACAAWwAoADEALAAgADQAMAApAF0ACgBoAHkAIAA9ACAAWwAoADEALAAgADQAMAApAF0ACgAKAG0AZQBzAGgAIAA9ACAAVABlAG4AcwBvAHIATQBlAHMAaAAoAFsAaAB4ACwAIABoAHkAXQApAAoAYABgAGAACgAKAEkAbgAgAHQAaABlACAAWwBdACgAIwBtAHkALQBwAHIAbwBnAHIAYQBtACkALAAgAHcAZQAgAGMAcgBlAGEAdABlACAAYQAgAG0AZQBzAGgAIABmAG8AcgAgAHMAaQBtAHUAbABhAHQAaQBvAG4AIAB1AHMAaQBuAGcAIABbAFMAaQBtAFAARQBHAF0AKABoAHQAdABwAHMAOgAvAC8AZABpAHMAYwByAGUAdABpAHoAZQAuAHMAaQBtAHAAZQBnAC4AeAB5AHoALwApAC4ACgA%3D), but maybe code should be on by default.

I think that because the code only gets numbered when there is a caption this is pretty safe -- that is it ISN'T every code example, just ones that are have captions!

Thoughts @fwkoch?

**Comment by kolibril13** [comment]
I am also pro turn code numbering on via default, because that would feel more coherent with tables and figures.

>  that is it ISN'T every code example, just ones that are have captions!

 analogue to this not every figure is numbered, only the ones with captions, as can be seen in this example

```

Hello [](#my-fig), and [](#my-program).

<!-- figure without number -->
![Alt text](wave.jpg)

<!-- numbered figure -->
:::{figure} wave.jpg
:name: my-fig
:alt: A drop of water.

A  water wave
:::


<!-- code without number -->
:::python
print(np.real(c))
print(np.imag(c))
:::

<!-- numbered code -->

:::{code-block} python
:name: my-program
:caption: Calculation with numpy.

print(np.real(c))
print(np.imag(c))
:::
```

**Comment by kolibril13** [comment]
Also, I've just tested the pdf export with code numbering on, but the code numbering does not appear in the pdf, as can be seen in the screenshot:

<kbd>
<img  src="https://user-images.githubusercontent.com/44469195/220148829-d5034173-2592-4a55-b00a-1ecdba27bcd2.png">
</kbd>


Code to reproduce this:

```
---
title: Hello

exports:
  - format: pdf
    template: arxiv_two_column
numbering:
  code: true 
---

+++ {"part": "abstract"}

Lorem ipsum ...
+++

:::python
print(np.real(c))
print(np.imag(c))
:::


:::{code-block} python
:name: my-program
:caption: Calculation with numpy.

print(np.real(c))
print(np.imag(c))
:::

```

**Comment by fwkoch** [comment]
I agree this is a good idea! Strange for `code` numbering to be different than everything else. (I was slightly confused at first because I was thinking about code line numbering, but this is just code block enumeration)

I think it's as simple as this: https://github.com/executablebooks/mystjs/pull/257

**Comment by rowanc1** [comment]
Released in myst-cli v0.1.17 🚀

I am going to leave this open, as I think there will actually have to be a few docs updates that we haven't done yet!

**Comment by rowanc1** [comment]
Had to fix up an iframe width issue, which meant I did this a bit sooner than I thought!!

[See numbering demo working in the sandbox!](https://myst-tools.org/sandbox?tab=demo&myst=YABgAGAAewBjAG8AZABlAC0AYgBsAG8AYwBrAH0AIABwAHkAdABoAG8AbgAKADoAbgBhAG0AZQA6ACAAbQB5AC0AcAByAG8AZwByAGEAbQAKADoAYwBhAHAAdABpAG8AbgA6ACAAQwByAGUAYQB0AGkAbgBnACAAYQAgAFQAZQBuAHMAbwByAE0AZQBzAGgAIAB1AHMAaQBuAGcAIABTAGkAbQBQAEUARwAKAGYAcgBvAG0AIABkAGkAcwBjAHIAZQB0AGkAegBlACAAaQBtAHAAbwByAHQAIABUAGUAbgBzAG8AcgBNAGUAcwBoAAoACgBoAHgAIAA9ACAAWwAoADEALAAgADQAMAApAF0ACgBoAHkAIAA9ACAAWwAoADEALAAgADQAMAApAF0ACgAKAG0AZQBzAGgAIAA9ACAAVABlAG4AcwBvAHIATQBlAHMAaAAoAFsAaAB4ACwAIABoAHkAXQApAAoAYABgAGAACgAKAEkAbgAgAHQAaABlACAAWwBdACgAIwBtAHkALQBwAHIAbwBnAHIAYQBtACkALAAgAHcAZQAgAGMAcgBlAGEAdABlACAAYQAgAG0AZQBzAGgAIABmAG8AcgAgAHMAaQBtAHUAbABhAHQAaQBvAG4AIAB1AHMAaQBuAGcAIABbAFMAaQBtAFAARQBHAF0AKABoAHQAdABwAHMAOgAvAC8AZABpAHMAYwByAGUAdABpAHoAZQAuAHMAaQBtAHAAZQBnAC4AeAB5AHoALwApAC4A)
